src/core/Directus/Database/TableGateway/BaseTableGateway.php

```diff
@@ -398,9 +398,26 @@ public function addRecordByArray(array $recordData)
         if ($columnObject->hasAutoIncrement()) {
             $recordData[$primaryKey] = $TableGateway->getLastInsertValue();
         }
+        else if ($columnObject->hasPrimaryKey())
+        {
+            // identify the autoincrement field+value to retrieve the new record
+            $fieldAutoIncrement = null;
+            foreach($this->getTableSchema()->getFields() as $f) {
+                if ($f->hasAutoIncrement()) {
+                    $fieldAutoIncrement = $f->getName();
+                    $lastInsertValue = $TableGateway->getLastInsertValue();
+                    break;
+                }
+            }
 
-        $columns = SchemaService::getAllNonAliasCollectionFieldNames($this->table);
+            // retrieve the full record
+            if (isset($fieldAutoIncrement)) {
+                $record = $this->findOneBy($fieldAutoIncrement, $lastInsertValue);
```

Comment:
Here I execute another query to retrieve the newly created record since the previous branch (`if $columnObject->hasAutoIncrement()`) can fail if `$primaryKey` is not also auto_increment